File: src/main/java/org/example/kalkulationsprogramm/repository/DokumentLockRepository.java
package org.example.kalkulationsprogramm.repository;

import java.util.Optional;

import org.example.kalkulationsprogramm.domain.DokumentLock;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface DokumentLockRepository extends JpaRepository<DokumentLock, Long> {

    Optional<DokumentLock> findByDokumentTypAndDokumentId(String dokumentTyp, Long dokumentId);

    void deleteByDokumentTypAndDokumentId(String dokumentTyp, Long dokumentId);
}

File: src/main/kotlin/org/example/kalkulationsprogramm/repository/DokumentLockRepository.kt
package org.example.kalkulationsprogramm.repository

import java.util.Optional
import org.example.kalkulationsprogramm.domain.DokumentLock
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface DokumentLockRepository : JpaRepository<DokumentLock, Long> {
    fun findByDokumentTypAndDokumentId(dokumentTyp: String, dokumentId: Long?): Optional<DokumentLock>

    fun deleteByDokumentTypAndDokumentId(dokumentTyp: String, dokumentId: Long?)
}
